src/language/class-modifiers.md

```diff
@@ -0,0 +1,240 @@
+  implements Vehicle {
+    @override
+    void moveForward …
+}
+```
+
+Disallowing inheritance makes `interface` useful for:
+- Defining API interfaces.
+- Preventing users of the API from overriding a subset of methods.
+
+## `final` 
+
+Use the `final` modifier to **disallow extending _and_ implementing** a class.
+Final classes can be constructed.
+
+```dart
+final class Vehicle {
+  String make;
+  String model;
+  void moveForward(int meters) { … }
+}
+
+var myCar = Vehicle(     // Can be constructed
+  make: 'Ford',
+  model: 'T',
+);
+
+class Car extends     // ERROR: Cannot be inherited
+  Vehicle {
+    int passengers;
+…
+}
+
+class MockVehicle     // ERROR: Cannot be implemented     
+  implements Vehicle {
+    @override
+    void moveForward …
+}
+```
+
+Disallowing inheritance and implementation makes `final` useful for:
+- Closing the type hierarchy.
+- Ensuring that adding new members to the class won't risk breaking changes for 
+its consumers.
+- Ensuring access to private members on instances of the final class will never
+throw a [`noSuchMethod`][] error.
+
+## `sealed`
+
+Use the `sealed` modifier to define a class that **cannot be constructed, extended,
+or implementated** outside of the library where the class is defined. 
+
+The purpose of the `sealed` modifier is to enable [_exhaustiveness checking_][]
+for switch cases.
+Any possible direct subtypes of a sealed type can only exist in the same library,
+so the compiler is always aware of the extent of a sealed type's hierarchy.
+
+This allows the compiler to alert you when a switch is not exhaustively handling
+all possible subtypes in its cases.
+
+```dart
+sealed class Vehicle {
+  String make;
+  String model;
+  void moveForward(int meters) { … }
+}
+
+class Car extends Vehicle { }
+class Truck implements Vehicle { }
+class Bicycle extends Vehicle { }
+
+// ...
+
+return switch (vehicle) {     // ERROR: The switch is missing a subtype of Vehicle
+  Car => 'vroom',
+  Truck => 'VROOOOMM'
+};
+```
+
+## `abstract`
+
+Use the `abstract` modifier to define a class that can’t be constructed.
+Abstract classes are useful for defining interfaces, often with some
+implementation. If you want your abstract class to appear to be instantiable,
+define a [factory constructor][].
+
+Abstract classes often have [abstract methods][].
+Here’s an example of declaring an abstract class that has an abstract
+method:
+
+<?code-excerpt "misc/lib/language_tour/classes/misc.dart (abstract)"?>
+```dart
+// This class is declared abstract and thus
+// can't be instantiated.
+abstract class AbstractContainer {
+  // Define constructors, fields, methods...
+
+  void updateChildren(); // Abstract method.
+}
+```
+
+## `class`, `mixin`, or `mixin class`?
+
+The `mixin` declaration defines a [mixin][]. The `mixin` _modifier_ can only be
+applied to a `class` declaration to create a _mixin class_: a class that can be
+treated both as a class and a mixed in. Regular classes cannot be mixed in.
+
```

Comment:
But also, maybe an example of a mixin class would be useful? Like how we do with the other modifiers.